### ISSUE TITLE
python3.6 maximum recursion depth exceeded

### ISSUE THREAD

**Issue by DanAlbert** [PyCQA/pylint#2591]
This might be related to #2388, but I'm unsure. This doesn't reproduce with Python 3.5. Not sure if it matters, but I'm on Debian.

### Steps to reproduce
1. Clone https://github.com/DanAlbert/pylint-recursion-exceeded-repro-case
2. `bash run.sh`

### Current behavior
> RecursionError: maximum recursion depth exceeded

Full log here: https://github.com/DanAlbert/pylint-recursion-exceeded-repro-case/blob/master/log

### Expected behavior
Pylint runs successfully

### pylint --version output
```bash
$ pylint --version
pylint 2.1.1
astroid 2.1.0-dev
Python 3.6.2 (default, Aug  4 2017, 14:35:04)
[GCC 6.3.0 20170516]
```

**Comment by PCManticore** [comment]
Oh that's bad, thanks for creating an issue, folks!

**Comment by DanAlbert** [comment]
It's surprisingly simple to work around. If my repro case is simplified any more (even keeping the same logic, just removing an intermediate) it no longer crashes.

**Comment by PCManticore** [comment]
This seems to have been fixed with the release of `pylint 2.4.3` and `astroid 2.3.2`. Can you give it a go and let us know if you can still reproduce the issue?

**Comment by DanAlbert** [comment]
Yep, the repro case no longer crashes. Thanks!
